### PR TITLE
fix: Generate https IDs for entities instead of file://

### PR DESCRIPTION
**Pull request by Panaetius**
Entity id's are now `http://<host>/blob/<sha>/<path>` instead of `file://blob/<sha>/<path>`

Also changes `Activity` ids to use the remote host if it's available for their ids.

### CODE REVIEW

**Comment by rokroskar** [comment]
Activities still have `https://localhost`, e.g.:

```
<https://localhost/activities/commit/2e7a7892ea83a3487cab7d262caa2e376f2e9318/outputs/output_stdout> <http://www.w3.org/ns/prov#activity> <https://localhost/activities/commit/2e7a7892ea83a3487cab7d262caa2e376f2e9318> .
```

Do you want to fix that here as well?

**Comment by Panaetius** [comment]
> Activities still have `https://localhost`, e.g.:
> 
> ```
> <https://localhost/activities/commit/2e7a7892ea83a3487cab7d262caa2e376f2e9318/outputs/output_stdout> <http://www.w3.org/ns/prov#activity> <https://localhost/activities/commit/2e7a7892ea83a3487cab7d262caa2e376f2e9318> .
> ```
> 
> Do you want to fix that here as well?

Yes makes sense to add the whole logic with env value, remote value or localhost there as well, I'll do it in this PR

**Comment by jachro** [comment]
I've just verified the fix on renku-kuba env and can say it looks the IDs are good now:
```
{
    "data": {
        "lineage": {
            "nodes": [
                {
                    "id": "https://renku-kuba.dev.renku.ch/blob/d54379107bb1767e1ea8647ab0282b66e8d683eb/src/00-FilterFlights.py",
                    "label": "src/00-FilterFlights.py@d54379107bb1767e1ea8647ab0282b66e8d683eb",
                    "type": "File"
                },
                {
                    "id": "https://renku-kuba.dev.renku.ch/blob/9dbf1f926cb397bb2fa886afde4a106e00b828c0/data/output/2019-01-flights-count.txt",
                    "label": "data/output/2019-01-flights-count.txt@9dbf1f926cb397bb2fa886afde4a106e00b828c0",
                    "type": "File"
                },
                {
                    "id": "https://renku-kuba.dev.renku.ch/activities/commit/b0ee3b05e1c82bfbcd1cc8e9eb821a48767802b2",
                    "label": "renku run papermill notebooks/01-CountFlights.ipynb notebooks/01-CountFlights.ran.ipynb -p input_path data/output/2019-01-flights-filtered.csv -p output_path data/output/2019-01-flights-count.txt",
                    "type": "ProcessRun"
                },
                {
                    "id": "https://renku-kuba.dev.renku.ch/blob/9dbf1f926cb397bb2fa886afde4a106e00b828c0/data/output/2019-01-flights-filtered.csv",
                    "label": "data/output/2019-01-flights-filtered.csv@9dbf1f926cb397bb2fa886afde4a106e00b828c0",
                    "type": "File"
                },
                {
                    "id": "https://renku-kuba.dev.renku.ch/blob/c9b6e6b76206de4b1153a87cd3960eb85256c1c6/data/201901_us_flights_1/2019-01-flights.csv.zip",
                    "label": "data/201901_us_flights_1/2019-01-flights.csv.zip@c9b6e6b76206de4b1153a87cd3960eb85256c1c6",
                    "type": "File"
                },
                {
                    "id": "https://renku-kuba.dev.renku.ch/blob/4bffecd2351476af10c49bb04dd26e9480d0a4ff/src/00-FilterFlights.py",
                    "label": "src/00-FilterFlights.py@4bffecd2351476af10c49bb04dd26e9480d0a4ff",
                    "type": "File"
                },
                {
                    "id": "https://renku-kuba.dev.renku.ch/blob/b0ee3b05e1c82bfbcd1cc8e9eb821a48767802b2/data/output/2019-01-flights-count.txt",
                    "label": "data/output/2019-01-flights-count.txt@b0ee3b05e1c82bfbcd1cc8e9eb821a48767802b2",
                    "type": "File"
                },
                {
                    "id": "https://renku-kuba.dev.renku.ch/activities/commit/9dbf1f926cb397bb2fa886afde4a106e00b828c0/steps/step_2",
                    "label": "renku update",
                    "type": "ProcessRun"
                },
                {
                    "id": "https://renku-kuba.dev.renku.ch/blob/7d8806ca9e223a4ab7d42fc6c32172de0783da6e/data/output/2019-01-flights-filtered.csv",
                    "label": "data/output/2019-01-flights-filtered.csv@7d8806ca9e223a4ab7d42fc6c32172de0783da6e",
                    "type": "File"
                },
                {
                    "id": "https://renku-kuba.dev.renku.ch/activities/commit/7d8806ca9e223a4ab7d42fc6c32172de0783da6e",
                    "label": "renku run python src/00-FilterFlights.py data/201901_us_flights_1/2019-01-flights.csv.zip data/output/2019-01-flights-filtered.csv",
                    "type": "ProcessRun"
                },
                {
                    "id": "https://renku-kuba.dev.renku.ch/activities/commit/9dbf1f926cb397bb2fa886afde4a106e00b828c0/steps/step_1",
                    "label": "renku update",
                    "type": "ProcessRun"
                },
                {
                    "id": "https://renku-kuba.dev.renku.ch/blob/622ca40e4e88e348160786597f0df00481631248/notebooks/01-CountFlights.ipynb",
                    "label": "notebooks/01-CountFlights.ipynb@622ca40e4e88e348160786597f0df00481631248",
                    "type": "File"
                }
            ],
            "edges": [
                {
                    "source": "https://renku-kuba.dev.renku.ch/blob/622ca40e4e88e348160786597f0df00481631248/notebooks/01-CountFlights.ipynb",
                    "target": "https://renku-kuba.dev.renku.ch/activities/commit/b0ee3b05e1c82bfbcd1cc8e9eb821a48767802b2"
                },
                {
                    "source": "https://renku-kuba.dev.renku.ch/blob/c9b6e6b76206de4b1153a87cd3960eb85256c1c6/data/201901_us_flights_1/2019-01-flights.csv.zip",
                    "target": "https://renku-kuba.dev.renku.ch/activities/commit/7d8806ca9e223a4ab7d42fc6c32172de0783da6e"
                },
                {
                    "source": "https://renku-kuba.dev.renku.ch/blob/4bffecd2351476af10c49bb04dd26e9480d0a4ff/src/00-FilterFlights.py",
                    "target": "https://renku-kuba.dev.renku.ch/activities/commit/9dbf1f926cb397bb2fa886afde4a106e00b828c0/steps/step_1"
                },
                {
                    "source": "https://renku-kuba.dev.renku.ch/blob/9dbf1f926cb397bb2fa886afde4a106e00b828c0/data/output/2019-01-flights-filtered.csv",
                    "target": "https://renku-kuba.dev.renku.ch/activities/commit/9dbf1f926cb397bb2fa886afde4a106e00b828c0/steps/step_2"
                },
                {
                    "source": "https://renku-kuba.dev.renku.ch/activities/commit/7d8806ca9e223a4ab7d42fc6c32172de0783da6e",
                    "target": "https://renku-kuba.dev.renku.ch/blob/7d8806ca9e223a4ab7d42fc6c32172de0783da6e/data/output/2019-01-flights-filtered.csv"
                },
                {
                    "source": "https://renku-kuba.dev.renku.ch/activities/commit/9dbf1f926cb397bb2fa886afde4a106e00b828c0/steps/step_2",
                    "target": "https://renku-kuba.dev.renku.ch/blob/9dbf1f926cb397bb2fa886afde4a106e00b828c0/data/output/2019-01-flights-count.txt"
                },
                {
                    "source": "https://renku-kuba.dev.renku.ch/blob/d54379107bb1767e1ea8647ab0282b66e8d683eb/src/00-FilterFlights.py",
                    "target": "https://renku-kuba.dev.renku.ch/activities/commit/7d8806ca9e223a4ab7d42fc6c32172de0783da6e"
                },
                {
                    "source": "https://renku-kuba.dev.renku.ch/activities/commit/b0ee3b05e1c82bfbcd1cc8e9eb821a48767802b2",
                    "target": "https://renku-kuba.dev.renku.ch/blob/b0ee3b05e1c82bfbcd1cc8e9eb821a48767802b2/data/output/2019-01-flights-count.txt"
                },
                {
                    "source": "https://renku-kuba.dev.renku.ch/blob/622ca40e4e88e348160786597f0df00481631248/notebooks/01-CountFlights.ipynb",
                    "target": "https://renku-kuba.dev.renku.ch/activities/commit/9dbf1f926cb397bb2fa886afde4a106e00b828c0/steps/step_2"
                },
                {
                    "source": "https://renku-kuba.dev.renku.ch/activities/commit/9dbf1f926cb397bb2fa886afde4a106e00b828c0/steps/step_1",
                    "target": "https://renku-kuba.dev.renku.ch/blob/9dbf1f926cb397bb2fa886afde4a106e00b828c0/data/output/2019-01-flights-filtered.csv"
                },
                {
                    "source": "https://renku-kuba.dev.renku.ch/blob/c9b6e6b76206de4b1153a87cd3960eb85256c1c6/data/201901_us_flights_1/2019-01-flights.csv.zip",
                    "target": "https://renku-kuba.dev.renku.ch/activities/commit/9dbf1f926cb397bb2fa886afde4a106e00b828c0/steps/step_1"
                },
                {
                    "source": "https://renku-kuba.dev.renku.ch/blob/7d8806ca9e223a4ab7d42fc6c32172de0783da6e/data/output/2019-01-flights-filtered.csv",
                    "target": "https://renku-kuba.dev.renku.ch/activities/commit/b0ee3b05e1c82bfbcd1cc8e9eb821a48767802b2"
                }
            ]
        }
    }
}
```

File: renku/core/models/jsonld.py [hunk 1]
```diff
@@ -256,10 +256,20 @@ def _propagate_reference_contexts(
...
+
+                prefix, suffix = subtype.split(':', 1)
+
+                if prefix in merge_ctx:
+                    expanded_subtype = '{}{}'.format(merge_ctx[prefix], suffix)
+
                 current_context['@context'].append({
                     fullname(cls) + '_' + subtype: {
-                        '@id': subtype,
+                        '@id': expanded_subtype,
```
Comment (rokroskar) [inline renku/core/models/jsonld.py]: 🙈

**Pull request merged**